roles/openshift_master/tasks/main.yml

```diff
@@ -23,7 +23,7 @@
       api_port: "{{ openshift_master_api_port | default(None) }}"
       api_url: "{{ openshift_master_api_url | default(None) }}"
       api_use_ssl: "{{ openshift_master_api_use_ssl | default(None) }}"
-      public_api_url: "{{ openshift_master_public_api_url | default(None) }}"
+      public_api_url: "{{ openshift_master_public_api_url | default('https://' ~ openshift.common.public_ip ~ ':8443') }}"
```

Comment:
Good. Now I know where theses variables get their values ! Thank you

So I tested with `openshift_master_public_api_url` defaulting to `None` and this is working properly.

However I tested by commenting infra-node related code (only for GCE) because currently this is not working neither for 0 or more infra-nodes.

I suppose that nobody is using this code since it's not working. By commenting it until it's solved, that won't impact anyone.

PR https://github.com/openshift/openshift-ansible/pull/641 updated